### PR TITLE
Add Joy of Elixir

### PR DESCRIPTION
**Pull request by radar**
Joy of Elixir is an in-progress book to teach newbies Elixir that I'm writing.

Thanks to @ckhrysze for the prompting to add the book to this list.

### CODE REVIEW

File: free-programming-books.md [hunk 1]
```diff
@@ -1037,9 +1037,9 @@
 * [Elixir School](https://elixirschool.com) (HTML)
 * [Études for Elixir](http://chimera.labs.oreilly.com/books/1234000001642/index.html) (HTML)
 * [Getting Started Guide](http://elixir-lang.org/getting-started/introduction.html) (HTML) [(PDF, MOBI, EPUB)](https://github.com/potatogopher/elixir-getting-started)
+* [Joy of Elixir](https://joyofelixir.com) - (HTML)
 * [Learning Elixir](http://learningelixir.joekain.com) - Joseph Kain Blog (HTML)
 
-
```
Comment (eshellman) [inline free-programming-books.md]: removing this space upset the linter

File: free-programming-books.md [hunk 2]
```diff
@@ -1037,9 +1037,9 @@
 * [Elixir School](https://elixirschool.com) (HTML)
 * [Études for Elixir](http://chimera.labs.oreilly.com/books/1234000001642/index.html) (HTML)
 * [Getting Started Guide](http://elixir-lang.org/getting-started/introduction.html) (HTML) [(PDF, MOBI, EPUB)](https://github.com/potatogopher/elixir-getting-started)
+* [Joy of Elixir](https://joyofelixir.com) - (HTML)
```
Comment (eshellman) [inline free-programming-books.md]: Would adding an author name be appropriate? see #2383 . Also, #2357 suggests a way to mark "in-progress" books.

**Comment by radar** [comment]
Thanks for the feedback @eshellman. I've updated this PR now to cater to your feedback.

**Comment by eshellman** [comment]
Thanks for the PR, and for the book!

**Pull request merged**